Update WP CLI command to use indexation actions

herregroen opened the pull request:
## Context
<!--
What do we want to achieve with this PR? Why did we write this code?
-->

*

## Summary

<!--
Attach one of the following labels to the PR: `changelog: bugfix`, `changelog: enhancement`, `changelog: other`, `changelog: non-user-facing`.
If the changelog item is a bugfix, please use the following sentence structure: Fixes a bug where ... would ... (when ...).
If the changelog item is meant for the changelog of another repo, start you changelog item with the repo name between square brackets, for example: * [wordpress-seo-premium] Fixes a bug where ....
If the same changelog item is applicable to multiple changelogs/repos, add a separate changelog item for all of them.
-->
This PR can be summarized in the following changelog entry:

* Updates the `wp yoast indexable generate` command to use indexation actions and renames it to `wp yoast index`.

## Relevant technical choices:

* Also updates the indexation route to use a single more abstracted function to run indexations.

## Test instructions
<!--
Please follow these guidelines when creating test instructions:
- Please provide step-by-step instructions how to reproduce the issue, if applicable.
- Write step-by-step test instructions aimed at non-tech-savvy users, even if the PR is not user-facing.
-->
This PR can be tested by following these steps:

* Delete all your indexables.
* In wp cli run `wp yoast index` ( or in the docker setup `./wp.sh yoast index` ).

## UI changes
* [ ] This PR changes the UI in the plugin. I have added the 'UI change' label to this PR.

## Documentation
* [ ] I have written documentation for this change.

## Quality assurance

* [ ] I have tested this code to the best of my abilities
* [ ] I have added unittests to verify the code works as intended

Fixes #

johannadevos commented:
CR: ✅ 
- [x] A `get_limit` was added to all actions, and to the action interface.
- [x] The `index` command was renamed and rewritten.
- [x] The indexation routes per type are refactored.
- [x] The actions all have tests.
- [ ] The indexation action interface is not covered in tests, but also not ignored. But technically that problem already existed before this PR.

johannadevos commented:
Update: the dev architects decided that interfaces don't need to be tested, and also don't need a `@codeCoverageIgnore` tag.

igorschoester commented:
Acceptance singlesite ✅ 
Acceptance multisite:
* Main site ✅ 
* Other site 🚧 

Since that multisite issue, I decided to check it there. And it only works for the "main" site. It leaves the indexables of the other site empty.

Not sure this is really related to this issue and probably more of a separate issue. So if another issue is created for that, this can be merged.

herregroen commented:
I've just refactored the classes to be more inline with the WP CLI standards, they now match the documentation in https://make.wordpress.org/cli/handbook/commands-cookbook/ except you don't have to add them yourself but instead they register automatically under the namespace provided in the static method.

I've also added the `--network` option to run the indexation on all sites. Do note that this doesn't run migrations for all sites. That will be handled by https://github.com/Yoast/wordpress-seo/pull/14873
